### PR TITLE
fixes pickle mod

### PR DESCRIPTION
**Pull request by wyli**
Signed-off-by: Wenqi Li <wenqil@nvidia.com>

follow up of #3412 

### Description
```
======================================================================
ERROR: test_timing (__main__.IntegrationSegmentation3D)
----------------------------------------------------------------------
TypeError: can't pickle module objects

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/utils.py", line 497, in _wrapper
    raise RuntimeError(res.traceback) from res
RuntimeError: Traceback (most recent call last):
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/utils.py", line 449, in run_process
    output = func(*args, **kwargs)
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/utils.py", line 522, in _call_original_func
    return f(*args, **kwargs)
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/test_integration_segmentation_3d.py", line 296, in test_timing
    self.train_and_infer(idx=3)
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/test_integration_segmentation_3d.py", line 258, in train_and_infer
    self.data_dir, device=self.device, cachedataset=idx, readers=_readers
  File "/home/jenkins/agent/workspace/MONAI-postmerge/monai_github/tests/test_integration_segmentation_3d.py", line 126, in run_training_test
    for batch_data in train_loader:
  File "/opt/conda/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 359, in __iter__
    return self._get_iterator()
  File "/opt/conda/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 305, in _get_iterator
    return _MultiProcessingDataLoaderIter(self)
  File "/opt/conda/lib/python3.6/site-packages/torch/utils/data/dataloader.py", line 918, in __init__
    w.start()
  File "/opt/conda/lib/python3.6/multiprocessing/process.py", line 105, in start
    self._popen = self._Popen(self)
  File "/opt/conda/lib/python3.6/multiprocessing/context.py", line 223, in _Popen
    return _default_context.get_context().Process._Popen(process_obj)
  File "/opt/conda/lib/python3.6/multiprocessing/context.py", line 284, in _Popen
    return Popen(process_obj)
  File "/opt/conda/lib/python3.6/multiprocessing/popen_spawn_posix.py", line 32, in __init__
    super().__init__(process_obj)
  File "/opt/conda/lib/python3.6/multiprocessing/popen_fork.py", line 19, in __init__
    self._launch(process_obj)
  File "/opt/conda/lib/python3.6/multiprocessing/popen_spawn_posix.py", line 47, in _launch
    reduction.dump(process_obj, fp)
  File "/opt/conda/lib/python3.6/multiprocessing/reduction.py", line 60, in dump
    ForkingPickler(file, protocol).dump(obj)
TypeError: can't pickle module objects



```
### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/data/utils.py [hunk 1]
```diff
@@ -76,8 +76,11 @@
     "pad_list_data_collate",
     "no_collation",
     "convert_tables_to_dicts",
+    "SUPPORTED_PICKLE_MOD",
 ]
 
+SUPPORTED_PICKLE_MOD = {"pickle": pickle}
```
Comment (Nic-Ma) [inline monai/data/utils.py:82]: Thanks for the quick fix for the issue.
In this way, I feel we can't support any other pickle tool unless hard-code here first, other pickle tool should be available in PyTorch `torch.save`.

Thanks.

Comment (wyli) [reply monai/data/utils.py:82]: Hi @Nic-Ma, I've checked that it is possible to edit the dictionary from the user side:
```py
>>> from monai.data.dataset import PersistentDataset
>>> PersistentDataset([1, 2, 3], lambda x: x, cache_dir=None, pickle_module="test")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Documents/MONAI/monai/data/dataset.py", line 185, in __init__
    self.pickle_module = look_up_option(pickle_module, SUPPORTED_PICKLE_MOD)
  File "/Users/Documents/MONAI/monai/utils/module.py", line 121, in look_up_option
    raise ValueError(f"Unsupported option '{opt_str}', " + supported_msg)
ValueError: Unsupported option 'test', Available options are {'pickle'}.

>>> import pickle
>>> from monai.data import utils
>>> utils.SUPPORTED_PICKLE_MOD["test"] = pickle
>>> PersistentDataset([1, 2, 3], lambda x: x, cache_dir=None, pickle_module="test")
<monai.data.dataset.PersistentDataset object at 0x7fd5b809fe50>

```

**Review by Nic-Ma — APPROVED** [review]
Looks good to me.
Could you please help add doc-string somewhere to show 2 points:
1. How to set 3rd party pickle tool as your example.
2. Why we need to use string "pickle" instead of modules directly due to multi-processing issue.

Thanks.

**Pull request merged**